docs/content/examples/grid/SearchingAndFiltering.js

```diff
@@ -0,0 +1,387 @@
+import { Content, Controller, TreeAdapter } from "cx/ui";
+import { Md } from "../../../components/Md";
+import { CodeSplit } from "../../../components/CodeSplit";
+import { CodeSnippet } from "../../../components/CodeSnippet";
+import { Grid, Tab, TextField, TreeNode } from "cx/widgets";
+import { casual } from "../data/casual";
+import { getSearchQueryPredicate } from "cx/util";
+
+class PageController extends Controller {
+    onInit() {
+        this.store.set(
+            "$page.employees",
+            Array.from({ length: 10 }).map((r, i) => ({
+                id: i + 1,
+                fullName: casual.full_name,
+                phone: casual.phone,
+                city: casual.city
+            }))
+        );
+
+        this.store.set("$page.data", this.generateData());
+
+        // Allow filtering by name, phone, and city
+        this.addTrigger(
+            "filter",
+            ["$page.search"],
+            (search) => {
+                if (!search) {
+                    this.store.set("$page.filteredData", this.store.get("$page.data"));
+                    return;
+                }
+                const tree = this.store.get("$page.data");
+                const filtered = filterTree(tree, search.toLowerCase());
+                this.store.set("$page.filteredData", filtered);
+            },
+            true
+        );
+    }
```

Comment:
Wouldn't it be easier to use the `onCreateFilter` here? This looks pretty scary.